docs/design/2022-12-01-enrich-the-information-of-explain-analyze-for-tablescan.md

```diff
@@ -0,0 +1,81 @@
+# Enrich the information of Explain Analyze for TableScan
+
+- Author: [Yunyan Hong](https://github.com/hongyunyan)
+
+## Introduction
+
+Currently `Explain Analyze` show limited exection infos when SQL runs on TiFlash, mainly including:
+
+- time: the time cost by executing this operator
+- loops: the count of the blocks read by this operator, that is,  the number of times stream->read() is executed
+- threads: The concurrency degree of this step, that is, the number of streams calculated at the same time
+- actRows: The number of rows returned by the operator
+
+![overview](./images/2022-12-01-enrich-the-information-of-explain-analyze-for-tablescan-1.png)
+
+For the storage layer of TiFlash, `Explain Analyze` misses a lot of important and meaningful information, such as
+
+- The number of rows/packs in DMFiles scanned in the `TableFullScan` operator
+- The number of rows/packs in DMFiles skipped by the rough set index in `TableFullScan` operator
+- The time to load rough set index in `TableFullScan` operator
+- The time to read all related DMFiles in `TableFullScan` operator
+- The time to create snapshots in `TableFullScan` operator
```

Comment:
What about `TableRangeScan`?

Reply:
Is `TableRangeScan` brought into tidb recently? What's the difference between `TableFullScan` and `TableRangeScan`?
Usually, I only see `TableFullScan` in the `explain analyze ...` result

Reply:
TableRangeScan is always there...
TableRangeScan is triggered when the where condition contains a primary key, both TableFullScan and TableRangeScan should be handled the same way on the TiFlash side.
```
mysql> CREATE TABLE test.test (
    ->   `pkey` smallint(28),
    ->   PRIMARY KEY CLUSTERED (pkey)
    -> );
Query OK, 0 rows affected (0.08 sec)

mysql> ALTER TABLE test.test SET TIFLASH REPLICA 1;
Query OK, 0 rows affected (0.08 sec)

mysql> 
mysql> 
mysql> 
mysql> explain select * from test where pkey > 0;
+---------------------------+---------+--------------+---------------+------------------------------------------------+
| id                        | estRows | task         | access object | operator info                                  |
+---------------------------+---------+--------------+---------------+------------------------------------------------+
| TableReader_12            | 3333.33 | root         |               | data:ExchangeSender_11                         |
| └─ExchangeSender_11       | 3333.33 | mpp[tiflash] |               | ExchangeType: PassThrough                      |
|   └─TableRangeScan_10     | 3333.33 | mpp[tiflash] | table:test    | range:(0,+inf], keep order:false, stats:pseudo |
+---------------------------+---------+--------------+---------------+------------------------------------------------+
```

Reply:
like region based pruning

Reply:
When the condition in where is about primary key, it will be TableRangeScan instead of TableFullScan